### ISSUE TITLE
Use multiples attribute exclusions when filtering nodes

### ISSUE THREAD

**Issue by fgutierrezz** [rundeck/rundeck#4372]
**Is your feature request related to a problem? Please describe.**
When you create a job and want to use different conditions to filter your nodes is not allowed to use more than one attribute to exclude nodes in a Job. 

Example: One exclusion and one inclusion ( this works )
 !tags: snp.* hostname:oracle_db

Example : several attribute exclusions ( this doesn't work )

 !tags: snp.* !hostname: oracle_db !nodename: mysql_db 

**Describe the solution you'd like**
it may be good if you can use multiple attributes to exclude nodes. 

**Describe alternatives you've considered**
-
**Additional context**
When using two or more exclusions, only the first exclusion is working. 

![captura de pantalla de 2019-01-08 21-15-00](https://user-images.githubusercontent.com/30966360/50934188-afba5000-1446-11e9-9619-8d5eea10872c.png)
![captura de pantalla de 2019-01-08 21-22-31](https://user-images.githubusercontent.com/30966360/50934189-b052e680-1446-11e9-858a-adae6f6d8a3e.png)
![captura de pantalla de 2019-01-08 21-31-04](https://user-images.githubusercontent.com/30966360/50934190-b052e680-1446-11e9-8b1a-c27e98590977.png)

**Comment by diogokiss** [comment]
I'm facing the same issue with Rundeck version 2.6.11.

**Comment by Ultimation** [comment]
Also seeing this issue on 3.0.22

**Comment by fardin01** [comment]
Hi @fdevans. In refernce to https://github.com/rundeck/rundeck/issues/3403#issuecomment-611597978, we are experiencing this issue in version 3.3.2. Any chance this can be looked into? :)

**Comment by Vebryn** [comment]
Still occuring with job API, neither with url parameter, neither with json data :

`curl -XPOST -b"JSESSIONID=node0lu48m62ka0f6yo5dgody7wam8.node0; Path=/; HttpOnly" "http://localhost:4440/api/1/job/79c369ee-30a1-4875-ba6e-f7865209602c/run" --data-urlencode 'filter=adresse_ip: 10.* !nodename: fleur-0'`

`curl -H "Content-Type: application/json" -XPOST -b"JSESSIONID=node0lu48m62ka0f6yo5dgody7wam8.node0; Path=/; HttpOnly" "http://localhost:4440/api/1/job/79c369ee-30a1-4875-ba6e-f7865209602c/run" -d '{ filter: "adresse_ip: 10.* !codesite: 10536" }'`

Both queries run a job with all nodes (like with a .* filter). "!" char doesn't appear on execution definition. 

`Include nodes matching: adresse_ip: 10.* codesite: 10536 `

 "!" appear on arrow link.

`https://rundeck/project/mon_projet/nodes?filter=adresse_ip%3A+10.*+%21codesite%3A+10536`

**Comment by binaryfire** [comment]
I just discovered this myself. Would be great to get this working- we rely heavily on tags and not being able to exclude more than one is problematic. I'm surprised this is the default behaviour. If we're excluding the localhost (which we do for most jobs) it means we're unable to define any other exclusion patterns.

Has anyone found a good workaround for this?

**Comment by jclendineng** [comment]
Any word on this? We use tags for everything and it's somewhat useless if you can only exclude 1 tag.

**Comment by jjblack** [comment]
> Any word on this? We use tags for everything and it's somewhat useless if you can only exclude 1 tag.

I'm able to exclude multiple tags (v4.6.1) with the exclude filter. But I am not able to exlcude by multiple attributes (eg. tags and a nodename)

**Comment by binaryfire** [comment]
Not sure the “enhancement” tag on this issue is correct. Seems like a bug / broken functionality to me.

Tagging @gschueler. Could you take a look at this? It’s been open since 2019. Perhaps reclassifying it as a bug will help prioritise it?

**Comment by jclendineng** [comment]
> Not sure the “enhancement” tag on this issue is correct. Definitely seems like a bug / broken functionality to me.
> 
> 
> 
> Tagging @gschueler. Could you take a look at this? It’s been open since 2019. Perhaps reclassifying it as a bug will help prioritise it?
> 
> 
> 
> The title of this issue should really be “Excluding multiple attributes when filtering nodes doesn’t work”

I second this, the limitation seems arbitrary and the inability to filter more finely is pretty big.

**Comment by ibishop100** [comment]
Yes, this is a big deal really, we tend to get pretty big filters with tags, but it would be nice to also filter off facts pulled from the machine, or at least single hostnames.

**Comment by Crypter** [comment]
I just want to share that I'm still experiencing this is Rundeck v4.X
Is is still present in v5 ? I guess it is?

**Comment by fdevans** [comment]
Thanks for the report. As part of a cleanup of our issue tracker, we're closing issues older than 12 months, in line with Rundeck's support policy of one year from a release's date (see the [release calendar](https://docs.rundeck.com/docs/history/release-calendar.html) for currently supported versions). This isn't a judgment that the report is invalid — many of these are likely still real — but we can't verify against a version this old.

If you're still seeing this on a currently supported release (6.1.x or 5.20.x+), please open a new issue using our [bug report template](https://github.com/rundeck/rundeck/issues/new?template=bug_report.yml) with your version and reproduction steps. Please don't reopen this issue — a fresh issue keeps things tracked against a supported version and gets it in front of us properly. We're actively working on improving how issues get submitted and triaged going forward.

Thanks for your patience,
The Rundeck Team
